apps/drec-ui/src/device/pages/DetailViewPage/DetailViewPage.effects.ts

```diff
@@ -0,0 +1,105 @@
+import { useAllDeviceFuelTypes, useDeviceDetailData } from '../../../api';
+import { useParams } from 'react-router';
+import {
+    getDeviceAgeInYears,
+    getEnergyTypeImage,
+    getMainFuelType,
+    PowerFormatter
+} from '../../utils';
+import { EnergyTypeEnum, TUseDeviceDetailViewLogic } from '../../../utils';
+
+export const useDetailViewPageEffects = () => {
+    const { id } = useParams();
+
+    const { device, isLoading: isDeviceLoading } = useDeviceDetailData(+id);
+
+    const { allTypes, isLoading: isDeviceTypesLoading } = useAllDeviceFuelTypes();
+    const { locationProps, cardProps } = useDeviceDetailViewLogic({
+        device,
+        owner: device?.organizationId,
+        allTypes
+    });
+
+    const isLoading = isDeviceLoading || isDeviceTypesLoading;
+
+    return { locationProps, cardProps, device, isLoading, allTypes };
+};
+
+export const useDeviceDetailViewLogic: TUseDeviceDetailViewLogic = ({
+    device,
+    owner,
+    allTypes
+}) => {
+    const locationProps = {
+        // @should be changed to actual owner name
+        owner: `Device owner organization id ${owner}`,
+        location: `${device?.countryCode}`,
+        coordinates: `${device?.latitude}, ${device?.longitude}`
+    };
+
+    const { mainType, restType } = getMainFuelType(device?.fuelCode, allTypes);
+    const deviceIcon = getEnergyTypeImage(mainType.toLowerCase() as EnergyTypeEnum);
+
+    const cardProps = {
+        headingIconProps: {
+            icon: deviceIcon,
+            title: mainType,
+            subtitle: restType
+        },
+        specFields: [
+            {
+                label: 'Project name',
+                value: device?.projectName
+            },
+            {
+                label: 'D-REC ID',
+                value: device?.drecID
+            },
+            {
+                label: 'Status',
+                value: device?.status
+            },
+            {
+                label: 'Address',
+                value: device?.address
+            },
+            {
+                label: 'Zip Code',
+                value: device?.zipCode
+            },
+            {
+                label: 'Installation',
+                value: device?.installationConfiguration
+            },
+            {
+                label: 'Capacity (MW)',
+                value: PowerFormatter.format(device?.capacity)
+            },
+            {
+                label: 'Age (Years)',
+                value: getDeviceAgeInYears(device?.commissioningDate)
+            },
+            {
+                label: 'Off Taker',
+                value: device?.offTaker
+            },
+            {
+                label: 'Sector',
+                value: device?.sector
+            },
+            {
+                label: 'Standard Compliance',
+                value: device?.standardCompliance
+            },
+            {
+                label: 'Labels',
+                value: device?.labels
+            }
```

Comment:
This part is not fitting into the Card component. Please check styling.
<img width="1223" alt="Screenshot 2021-09-03 at 11 59 32" src="https://user-images.githubusercontent.com/69903849/131980806-ea56798e-ecd5-462d-8611-60e22dae62ad.png">